### ISSUE TITLE
ipfs://CIDv0 does not work in Firefox 70

### ISSUE THREAD

**Issue by lidel** [ipfs-shipyard/ipfs-companion#815]
> Part of https://github.com/ipfs/ipfs/issues/337

Catching unhandled protocols is an experiment enabled by default:

![2019-11-15--12-52-39](https://user-images.githubusercontent.com/157609/68941689-de136500-07a6-11ea-9b1c-65c0fb5094d3.png)

### Issue with address bar / a href

`ipfs://QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR` 
gets lower-cased and produces an error in Firefox 70:

> ![Screenshot_2019-11-14 Issue with Protocol Handler](https://user-images.githubusercontent.com/157609/68874724-f8473780-0701-11ea-87b9-f2589c859cee.png)

### Issue with img src

```html
<img src="ipfs://QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR" />
```

Fails to load.

### Todo

- [x] see if it happens before or after handler in webextension
- [x] update error page to educate about case-sensitivity in native URLs and suggest  use of [CIDv1b32](https://github.com/ipfs/ipfs/issues/337)
  - idea: text field for pasting CIDv0 that does automatic conversion ?

cc https://github.com/ipfs-shipyard/ipfs-companion/issues/527

**Comment by andrew** [comment]
I ran into this today, the display of the fix in #824 being in a red box didn't help me understand the fix as I skimmed right over that, only after went to report an issue, searched and found this then went back and looked again did I realize what the page was saying, may need some tweaking of the display and/or text? cc @jessicaschilling

**Comment by jessicaschilling** [comment]
@andrew -- thanks for raising this. Agreed that we can make the text in the red box more specific and helpful (and therefore less likely to be ignored). I'm currently working through a crib sheet of miscellaneous Companion UI fixes and will add it to that and name-check this issue when the changes makes its way into a PR.

**Comment by jessicaschilling** [comment]
@lidel -- Sorry to ask for the hint, but can I check in on the best way of addressing this when you have the chance? Unless I'm missing something, sounds like the best approach is just to modify the error page and then point Companion to the new page's CID. True?

**Comment by lidel** [comment]
@jessicaschilling Correct. 

This error page is loaded from IPFS because of how Firefox `protocol_handler` API works – this entire issue is Firefox-specific – the redirect has to be to a remote Origin, so I used public gateway.

If you want to tackle this, update CID with new version in those places:

- https://github.com/ipfs-shipyard/ipfs-companion/search?q=bafkreiewrj2pugsghd3flw2lk2fhvtmz6wipecnxep5qc5m3lfpf2mvjk4

**Comment by jessicaschilling** [comment]
@lidel and @andrew -- how about this?
https://gateway.ipfs.io/ipfs/QmTcoFD4WqmcT3uA7z3cF42uMRFJmnHZfPUjmdm8Du5gew

Local screenshots with fake URIs bashed in for the sake of illustration:
<img width="894" src="https://user-images.githubusercontent.com/1507828/87102656-c1351180-c20f-11ea-9f11-bebb4952d4f9.png">
<img width="894" src="https://user-images.githubusercontent.com/1507828/87102662-c4c89880-c20f-11ea-8e06-d714df449dbd.png">

@lidel, note inline favicon 😉 

**If either of you would be kind enough to look through the code for that page, would greatly appreciate -- once we're OK with the page I'll change the CID references in Companion in a new PR.**

**Comment by lidel** [comment]
Can we add  `ipns://` example to:

* A valid CIDv1 – this looks like `ipfs://{CIDv1}` or `ipns://{CIDv1}`

And shorten DNSLink one to:

* A valid DNSLink – this looks like `ipns://{domain-name-with-DNSLink}`

Otherwise LGTM! 
Feel free to open a PR that updates the CID everywhere :+1:

**Comment by jessicaschilling** [comment]
Updated at https://gateway.ipfs.io/ipfs/QmVGC4uCBDVEhCzsaJmvR5nVDgChM97kcYNehVm7L9jxtc ... PR to come 😊

**Comment by autonome** [comment]
@lidel can we autorewrite v0 to v1 in this case? Or do it in the "cannot process" page on client side, and have "click here to try v1 CID" instead of putting it on the user to do in CID Inspector?

**Comment by lidel** [comment]
@autonome we are unable to autofix – Firefox performs lowercasing before extension can read the CID, and the lowercasing is a lossy operation which we can't undo. 

We could have converter on the error page (instead of linking to https://cid.ipfs.io), 
but even then, the user would have to _manually_ paste the original CID/URI to input.
